routing/pathfind.go

```diff
@@ -62,6 +62,27 @@ type HopHint struct {
 	CLTVExpiryDelta uint16
 }
 
+// ShortHop describes the minimum of information that is needed for routing. It
+// is used as an input to construct a fully populated Route struct.
+type ShortHop struct {
```

Comment:
I like this addition, very clean to keep only the needed information around to create a route!

My main concern about this is that the number of different ways of representing a channel in the codebase is getting a bit out of hand... I think we Have `Hop`, `ChannelHop`, `Edge`, `ChannelEdgePolicy` and probably a few more. With the addition of `ShortHop` do you think there is a possibility to remove one of the existing ones, or reuse an existing one for this purpose?

Reply:
Yes, I agree. What I think is a problem at the moment, are those database-connected structs. This becomes especially visible in `SendToRoute`. Possibly a route is passed in that does not match what is in the database, but still there is no reason not to be able to properly perform the payment. So what we see happening for example is that we populate a `Route` structure and leave some fields empty that are not read anyway. 

In my opinion, the dependency on a db link or not should be made more explicit by using either a datastructure that always has the field populated or a datastructure that doesn't have the field at all.

I can imagine that when we start looking at the code in this way and slowly working out these db links where not needed, the number of data structures will reduce too.